docs: adding available variables and updating faq (#3927)
open-policy-agent/gatekeeper: website/docs/constrainttemplates.md

                  msg := sprintf("you must provide labels: %v", [missing])
                }
```

## Built-in variables across all engines

### Common variables

### Rego variables

| Variable | Description |
| --- | --- |
| `input.review` | Contains input request object under review |

> | `input.params` | Contains constraint parameters i.e. `input.params = constraint.spec` |

Author: I thought I wrote `input.parameters = constraint.spec.paramerets` but I mixed it up with CEL `params`. Fixed it!

> | `data.lib`     | Used to import rego code defined under `libs` in ConstraintTemplate |

Reviewer: ```suggestion
| `data.lib`     |  It serves as an import path for helper functions defined under `libs` in ConstraintTemplate, e.g. data.lib.exempt_container.is_exempt see [example](https://open-policy-agent.github.io/gatekeeper-library/website/validation/host-network-ports) |
```

> | `data.inventory` | Contains synced resources for referential policies |

Reviewer: ```suggestion
| `data.inventory` | Refers to a structure that stores synced cluster resources. It is used in Rego policies to validate or enforce referential rules based on the current state of the cluster. e.g. unique ingress host [example(https://open-policy-agent.github.io/gatekeeper-library/website/validation/uniqueingresshost/)|
```

### CEL variables

| Variable | Description |
| --- | --- |

> | `variables.params` | Contains constraint parameters i.e. `variable.params = constraint.spec` |

Reviewer: ```suggestion
| `variables.params` | Contains constraint parameters e.g. `variables.params.labels` see [example](https://open-policy-agent.github.io/gatekeeper-library/website/validation/requiredlabels) |
```

where do you see this i.e. `variable.params = constraint.spec`?

Author: It should be `constaint.spec.parameters`, changed it to your suggestion.

> | `variables.anyObject` | Contains either an object or (on DELETE requests) oldObject |

Reviewer: ```suggestion
| `variables.anyObject` | Contains either an object or (on DELETE requests) oldObject, see [example](https://open-policy-agent.github.io/gatekeeper-library/website/validation/requiredlabels) |
```
